util/ranger/detacher.go

```diff
@@ -194,7 +185,7 @@ func detachCNFCondAndBuildRangeForIndex(sctx sessionctx.Context, conditions []ex
 	res.Ranges = ranges
 	res.AccessConds = accessConds
 	res.RemainedConds = filterConds
-	res.EqCondCount = eqCount
+	res.EqCondCount = eqOrInCount
```

Comment:
`EqCondCount` is used when checking the physical property pushed down to data source.
We cannot let it also contain the count of `IN` function.

Reply:
`EqCondCount` is used in `getIndexCandidate` to check if a index path match a property and in `splitCorColAccessCondFromFilters` to extract cor-columns to push down.
It seems safe in these two place if we make `EqCondCount` contain `IN` expressions.